### ISSUE TITLE
YOLOV7 Neck hard code of output channels and loss weight.

### ISSUE THREAD

**Issue by lluo-Desktop** [open-mmlab/mmyolo#809]
### Prerequisite

- [X] I have searched [the existing and past issues](https://github.com/open-mmlab/mmyolo/issues) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmyolo.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version](https://github.com/open-mmlab/mmyolo).

### 🐞 Describe the bug

YOLOV7 Neck hard code of output channels :
Please check  file : mmyolo/models/necks/yolov7_pafpn.py  
                      function : def build_out_layer(self, idx: int) -> nn.Module:
                      line 200 : out_channels = self.out_channels[idx] * 2
The output channels of YOLOv7PAFPN will be double. 
This also can be found in yolov7 config (yolov7_l_syncbn_fast_8x16b-300e_coco.py) : 
                      Neck param : 
                      # The real output channel will be multiplied by 2
                      out_channels=[128, 256, 512],
                      Head Param : 
                      in_channels=[256, 512, 1024],
This can be very confusing.

YOLOV7 loss hard code of output channels :            
loss_cls=loss_cls_weight*(num_classes / 80 * 3 / num_det_layers)
loss_bbox=loss_bbox_weight * (3 / num_det_layers)
loss_obj=loss_obj_weight *((img_scale[0] / 640)**2 * 3 / num_det_layers)
The magic numbers (80, 3 , 3, 640) is also confusing.
                       

### Environment

TorchVision: 0.11.2
OpenCV: 4.7.0
MMEngine: 0.7.3
MMCV: 2.0.0rc4
MMDetection: 3.0.0rc6
MMYOLO: 0.5.0+
